exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java

```diff
@@ -131,6 +146,52 @@ public QueryResult run(final WorkManager workManager, final WebUserConnection we
     return new QueryResult(queryId, webUserConnection, webUserConnection.results);
   }
 
+  private void applyOptions(WebUserConnection webUserConnection) throws BadRequestException {
```

Comment:
This and the above can probably be simplified. Just keep the options as a string map until here. Then use `OptionManager.setLocalOption(String name, String value)`. (Why it is called "local" I don't know, nor do I know why we have one that takes an `Object` and another that takes a `String`, seems an unsafe overloading to me.) Anyway, that method does the very same string parsing this code does above, and avoids the need for the double type check.

Reply:
When the options come from the form, they do arrive as strings.  But if it is a JSON payload, then the options object values could be boolean, number, etc... any JSON value would be possible there.  I wanted to accept actual boolean / numeric values from a JSON payload rather than have them provide the booleans / number as string.